petclinic-frontend/src/features/Emailing/SendRawEmail/RawEmailPopUp.tsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from 'react';
+import { RawEmailModelDTO } from '@/features/Emailing/Model/RawEmailModel.ts';
+import { SendRawEmail } from '@/features/Emailing/Api/SendRawEmail.tsx';
+import './RawEmailCSS.css';
+
+const PopupWithForm: React.FC = () => {
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailTitle, setEmailTitle] = useState('');
+  const [body, setBody] = useState('');
+  const [statusCode, setStatusCode] = useState<number | null>(null);
+  const [statusMessage, setStatusMessage] = useState('');
+
+  // Function to open the popup
+  const openPopup = (): void => {
+    setIsPopupOpen(true);
+  };
+
+  // Function to close the popup
+  const closePopup = (): void => {
+    setIsPopupOpen(false);
+    // Reset status on closing
+    setStatusCode(null);
+    setStatusMessage('');
+  };
+
+  // Function to handle the body field and include "Enter" as a newline
+  const handleBodyChange = (
+    e: React.ChangeEvent<HTMLTextAreaElement>
+  ): void => {
+    setBody(e.target.value);
+  };
+
+  // Function to handle form submission
+  const handleSubmit = async (
+    e: React.FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    e.preventDefault();
+
+    // Prepare payload, converting newlines to <br />
+    const payload: RawEmailModelDTO = {
+      emailToSendTo: email,
+      emailTitle,
+      body: body.replace(/\n/g, '<br />'), // Replace newlines with <br />
```

Comment:
Ngl that's pretty cool

Reply:
Thank you! Found this online, proved to be very helpful since I didn't know how to transform breaks into <br>